dvc/remote/local/__init__.py

```diff
@@ -78,6 +78,7 @@ def __init__(self, repo, config):
             self.cache_types = types
         else:
             self.cache_types = copy(self.DEFAULT_CACHE_TYPES)
+        self.cache_type_confirmed = False
```

Comment:
For the record, this is a helper for the current behaviour, but in the future, we should detect supported link type more granularly, as you might have external local files on another fs than your repo workspace. But that would need to be handled later in a separate issue.

Reply:
Just to take note:
Solution for that might be storing map `(cache_device, path_info_device) -> working_link_type`
Devices could by obtained by `os.stat(path).st_dev`